app/src/main/java/com/internship/move/presentation/authentification/viewmodel/AuthenticationViewModel.kt

```diff
@@ -1,33 +1,68 @@
 package com.internship.move.presentation.authentification.viewmodel
 
-import android.content.ContentValues.TAG
-import android.util.Log
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.internship.move.data.model.User
+import com.internship.move.data.dto.user.*
 import com.internship.move.repository.UserRepository
+import com.internship.move.utils.extensions.toErrorResponse
+import com.squareup.moshi.JsonAdapter
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import java.io.File
 
 class AuthenticationViewModel(
-    private val repo: UserRepository
+    private val repo: UserRepository,
+    private val errorJsonAdapter: JsonAdapter<ErrorResponse>
 ) : ViewModel() {
 
+    val userRegisterData: MutableLiveData<UserRegisterResponseDto> = MutableLiveData()
```

Comment:
I suggest using backing fields for MutableLiveData variables. This means having the mutable variable be private and adding a LiveData public field that gets its value. This way the mutable liveData can be changed only from the ViewModel and you can observe the LiveData from outside.

```
private val _userRegisterData: MutableLiveData<UserRegisterResponseDto> = MutableLiveData()
    val userRegisterData: LiveData<UserRegisterResponseDto>
        get() = _userRegisterData
```